main.js

```diff
@@ -0,0 +1,203 @@
+url = 'https://raw.githubusercontent.com/alexsimkovich/patronage/main/api/data.json';
+
+let inCartItems = 0; //ile jest produktów w koszyku
+let productsInCart = []; //tablica do zapisywania ile jest danego produktu w koszyku
+let totalCost = 0; //koszt całkowity zamówienia
+
+
+fetch(url)
+    .then(function (response){
+        return response.json();
+    })
+    .then(function (data){
+        appendData(data);
+    })
+    .catch(function (err){
+        console.log(err)
+    });
+
+
+function appendData(data){ //załadowanie danych na stronie
+
+    let column1 = document.getElementById("col1");
+    let div;
+    let img;
+    let title;
+    let price;
+    let ingredients;
+    let button;
+
+    for(let i = 0; i < data.length; i++){
+
+        productsInCart.push({"inCart": 0});
+
+        div = document.createElement("div");
+        div.classList.add('product');
+
+        img = document.createElement('img');
+        img.src = data[i].image;
+        img.classList.add('product-image');
+        div.appendChild(img);
+
+        title = document.createElement("h3");
+        title.classList.add('title-product');
+        title.innerHTML = data[i].title;
+        div.appendChild(title);
+
+        price = document.createElement("h3");
+        price.classList.add('price-product');
+        price.innerHTML = data[i].price.toFixed(2); 
+        div.appendChild(price);
+
+        ingredients = document.createElement("h3");
+        ingredients.classList.add('ingredients-product');
+        ingredients.innerHTML = data[i].ingredients.join(", ");
+        div.appendChild(ingredients);
+
+        button = document.createElement("a");
+        button.classList.add('add-cart', 'cart-button');
+        button.innerHTML = "Zamów";
+        button.addEventListener('click', () => {
+            addToCart(data[i]);
+        })
+        
+        div.appendChild(button);
+        column1.appendChild(div);
+
+    }
+
+}
+
+function addToCart(product){ //funkcja, która wykonuje się po kliknięciu przycisku zamów
+
+    inCartItems++;
+    checkCart();
+    addTotalCost(product);
+    setItemInCart(product);
+}
+
+function checkCart(){ //sprawdzanie czy koszyk jest pusty
+    let cart = document.getElementById("col2");

```

Comment:
proszę przeczytać czym się różnią let od const. Naprzykład https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/